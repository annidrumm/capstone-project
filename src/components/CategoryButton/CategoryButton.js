import React from 'react'
import styled from 'styled-components/macro'
import forkIcon from '../../icons/fork.svg'
import gymIcon from '../../icons/gym.svg'
import beautyIcon from '../../icons/beauty.svg'
import fashionIcon from '../../icons/mannequin.svg'
import InteriorIcon from '../../icons/vase.svg'
import travelIcon from '../../icons/plane.svg'
import { useHistory } from 'react-router-dom'

export default function CategoryButton() {
  const history = useHistory()

  const routeChange = () => {
    let path = `InfluencerList`
    history.push(path)
  }

  return (
    <ContainerStyled>
      <section>
        <ButtonStyled category="food" onClick={routeChange}>
          <IconsStyled src={forkIcon} alt="fork and spoon" />
          <ButtonTextStyled>Food</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="fitness" onClick={routeChange}>
          <IconsStyled src={gymIcon} alt="dumbbell" />
          <ButtonTextStyled>Fitness</ButtonTextStyled>
        </ButtonStyled>
      </section>
      <section>
        <ButtonStyled category="beauty" onClick={routeChange}>
          <img src={beautyIcon} alt="lipsticks" />
          <ButtonTextStyled>Beauty</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="fashion" onClick={routeChange}>
          <img src={fashionIcon} alt="fashion mannequin" />
          <ButtonTextStyled>Fashion</ButtonTextStyled>
        </ButtonStyled>
      </section>
      <section>
        <ButtonStyled category="interior" onClick={routeChange}>
          <img src={InteriorIcon} alt="interior vase" />
          <ButtonTextStyled>Interior</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="travel" onClick={routeChange}>
          <img src={travelIcon} alt="plane" />
          <ButtonTextStyled>Travel</ButtonTextStyled>
        </ButtonStyled>
      </section>
    </ContainerStyled>
  )
}

const ContainerStyled = styled.section`
  margin-top: 70px;
`

const ButtonStyled = styled.button`
  width: 128px;
  height: 101px;
  background: transparent;
  margin: 10px;
  border-radius: 10%;
  cursor: pointer;
  border-color: ${({ category }) =>
    category === 'food'
      ? 'var(--darkyellow)'
      : category === 'fitness'
      ? 'var(--lightpink)'
      : category === 'beauty'
      ? 'var(--lightgreen)'
      : category === 'fashion'
      ? 'var(--lightblue)'
      : category === 'interior'
      ? 'var(--lightpurple)'
      : 'var(--darkyellow)'};
`
const IconsStyled = styled.img`
  margin-top: 15px;
  vertical-align: top;
`

const ButtonTextStyled = styled.h2`
  color: var(--font-color-white);
  font-size: 20px;
  font-weight: var(--font-weight-light);
  vertical-align: bottom;
  margin-bottom: 7px;
  margin-top: 3px;
`
