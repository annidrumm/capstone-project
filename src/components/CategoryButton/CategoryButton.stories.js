import React from 'react'
import CategoryButton from './CategoryButton'
import backIcon from '../../icons/back.svg'

export default {
  title: 'Components/Button',
  component: CategoryButton,
}

export const allCategories = () => {
  return <CategoryButton />
}

export const goBack = () => {
  return <img src={backIcon} alt="arrow left" />
}
