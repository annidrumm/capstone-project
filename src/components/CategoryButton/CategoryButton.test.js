import React from 'react'
import renderer from 'react-test-renderer'
import CategoryButton from './CategoryButton'
import '@testing-library/jest-dom/extend-expect'

describe('CategoryButton', () => {
  it('renders CategoryButton correctly', () => {
    const tree = renderer.create(<CategoryButton />)
    expect(tree).toMatchSnapshot()
  })
})
