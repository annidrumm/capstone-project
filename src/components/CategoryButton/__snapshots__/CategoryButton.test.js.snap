// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton renders CategoryButton correctly 1`] = `
<section
  className="CategoryButton__ContainerStyled-sc-13mtfqc-0 fVaACJ"
>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 dwCUFk"
      onClick={[Function]}
    >
      <img
        alt="fork and spoon"
        className="CategoryButton__IconsStyled-sc-13mtfqc-2 hEdiHb"
        src="fork.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Food
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 cTgMdW"
      onClick={[Function]}
    >
      <img
        alt="dumbbell"
        className="CategoryButton__IconsStyled-sc-13mtfqc-2 hEdiHb"
        src="gym.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Fitness
      </h2>
    </button>
  </section>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 ggLIMZ"
      onClick={[Function]}
    >
      <img
        alt="lipsticks"
        src="beauty.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Beauty
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 ivTBog"
      onClick={[Function]}
    >
      <img
        alt="fashion mannequin"
        src="mannequin.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Fashion
      </h2>
    </button>
  </section>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 gLpxBs"
      onClick={[Function]}
    >
      <img
        alt="interior vase"
        src="vase.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Interior
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 dwCUFk"
      onClick={[Function]}
    >
      <img
        alt="plane"
        src="plane.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Travel
      </h2>
    </button>
  </section>
</section>
`;

exports[`CategoryButton renders correctly 1`] = `
<section
  className="CategoryButton__ContainerStyled-sc-13mtfqc-0 fVaACJ"
>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 dwCUFk"
      onClick={[Function]}
    >
      <img
        alt="fork and spoon"
        className="CategoryButton__IconsStyled-sc-13mtfqc-2 hEdiHb"
        src="fork.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Food
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 cTgMdW"
      onClick={[Function]}
    >
      <img
        alt="dumbbell"
        className="CategoryButton__IconsStyled-sc-13mtfqc-2 hEdiHb"
        src="gym.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Fitness
      </h2>
    </button>
  </section>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 ggLIMZ"
      onClick={[Function]}
    >
      <img
        alt="lipsticks"
        src="beauty.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Beauty
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 ivTBog"
      onClick={[Function]}
    >
      <img
        alt="fashion mannequin"
        src="mannequin.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Fashion
      </h2>
    </button>
  </section>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 gLpxBs"
      onClick={[Function]}
    >
      <img
        alt="interior vase"
        src="vase.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Interior
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 dwCUFk"
      onClick={[Function]}
    >
      <img
        alt="plane"
        src="plane.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Travel
      </h2>
    </button>
  </section>
</section>
`;

exports[`CategoryButton renders subtractButton correctly 1`] = `
<section
  className="CategoryButton__ContainerStyled-sc-13mtfqc-0 fVaACJ"
>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 dwCUFk"
      onClick={[Function]}
    >
      <img
        alt="fork and spoon"
        className="CategoryButton__IconsStyled-sc-13mtfqc-2 hEdiHb"
        src="fork.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Food
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 cTgMdW"
      onClick={[Function]}
    >
      <img
        alt="dumbbell"
        className="CategoryButton__IconsStyled-sc-13mtfqc-2 hEdiHb"
        src="gym.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Fitness
      </h2>
    </button>
  </section>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 ggLIMZ"
      onClick={[Function]}
    >
      <img
        alt="lipsticks"
        src="beauty.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Beauty
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 ivTBog"
      onClick={[Function]}
    >
      <img
        alt="fashion mannequin"
        src="mannequin.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Fashion
      </h2>
    </button>
  </section>
  <section>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 gLpxBs"
      onClick={[Function]}
    >
      <img
        alt="interior vase"
        src="vase.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Interior
      </h2>
    </button>
    <button
      className="CategoryButton__ButtonStyled-sc-13mtfqc-1 dwCUFk"
      onClick={[Function]}
    >
      <img
        alt="plane"
        src="plane.svg"
      />
      <h2
        className="CategoryButton__ButtonTextStyled-sc-13mtfqc-3 iBKuEP"
      >
        Travel
      </h2>
    </button>
  </section>
</section>
`;
